### PR TITLE
[JENKINS-55787] Switch labels from entry to checkbox

### PR DESCRIPTION
**Pull request by jsoref**
[https://issues.jenkins-ci.org/browse/JENKINS-55787](https://issues.jenkins-ci.org/browse/JENKINS-55787)

Basically checkboxes should have labels, splitting the labels away from their checkboxes is poor UX, poor accessibility. This change brings the layout more inline with how normal settings UIs lay out checkboxes.

This supersedes #51

### CODE REVIEW

**Comment by alecharp** [comment]
Thank you @jsoref for your contribution, again :tada: !

I've made a screenshot without you fix:

2019-03-29-22-03-19_891x328

and with:

2019-03-29-22-05-57_887x329

to help me visualized the diff. I'm still not fluent in Jelly :wink:.

I do understand your point, having the checkbox before the label is normally what we expect. But I fear that in Jenkins, the way the configuration is visualized, it might be misunderstood. 

The overall configurations are "designed" like a `table`, with 
- a left column for the field label / title
- a right colum for the field value (with textbox, textarea, checkbox, etc.)

I fear that the modification of this "structure" might cause some incomprehension. 

I'm really sorry to engage the discusion a month and 1/2 after you opened this PR, this is a really big mistake on my side.

**Comment by oleg-nenashev** [comment]
@jsoref gentle ping, are you still interested in it?

**Comment by jsoref** [comment]
@alecharp: The table format for the configuration will be removed once I get enough of these changes merged. And this is one of countably few outliers.

**Comment by oleg-nenashev** [comment]
No strong opinion from me. We should be indeed moving towards the table-less UI, but right now it rather downgrades the UX. What would be the impact if we merge this change after the table removal PRs are merged?

**Comment by jsoref** [comment]
It looks a bit uglier in table-less. I'm happy to let some of these wait until we switch.

**Comment by jsoref** [comment]
Fwiw, divs landed in https://github.com/jenkinsci/jenkins/commit/87b72f57c3eaee2814fd63c12a9bc88b46015f28

**Review by oleg-nenashev — COMMENTED** [review]
On 2.277.1 RC it makes no differene:

Before and after:
image


The indent below the control is weird, but the patch does change that

File: src/main/resources/hudson/tasks/Mailer/global.jelly
Comment (jglick) [inline src/main/resources/hudson/tasks/Mailer/global.jelly:39]:
```suggestion
      <f:entry field="useSsl">
        <f:checkbox title="${%Use SSL}"/>
```

**Review by jglick — CHANGES_REQUESTED** [review]
`field` must remain on `f:entry` in order for help to work.

**Comment by jglick** [comment]
Is this still needed? It already looks OK to me in trunk:
mailer

File: src/main/resources/hudson/tasks/Mailer/global.jelly
Comment (jsoref) [reply src/main/resources/hudson/tasks/Mailer/global.jelly:42]:
```suggestion
      <f:entry field="useTls">
        <f:checkbox title="${%Use TLS}"/>
```

**Comment by jsoref** [comment]
I'm not sure if it's still needed. We turned off our jenkins instance a while ago and are now relying exclusively on GitHub for CI (we used Google Cloud build for a bit).

**Comment by alecharp** [comment]
Based on Jesse comment, I'm closing this for now. 
Thank you anyway @jsoref for the time you spent on this.

**Pull request closed without merging**
